### PR TITLE
Fn/refactoring

### PR DESCRIPTION
**Pull request by felipenoris**
This adds some love to the package, porting to Julia v0.6. After this, I can work on v1.0.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,9 @@
-*.cov
+*.jl.cov
+*.jl.*.cov
+*.jl.mem
+/deps/build.log
+/deps/deps.jl
+/deps/usr/
+/docs/build/
+/docs/site/
+/docs/out.xml
```
Comment (ScottPJones) [inline .gitignore]: Don't forget to have the trailing nl

File: .travis.yml [hunk 2]
```diff
@@ -2,16 +2,13 @@ language: julia
...
-    # - nightly
 services:
     - mongodb
 notifications:
     email: false
-sudo: false
-script:
-    - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
-    - julia -e 'Pkg.clone(pwd()); Pkg.build("Mongo"); Pkg.test("Mongo", coverage=true)'
+before_script:
```
Comment (ScottPJones) [inline .travis.yml:10]: These don't belong, comment out the script, and use the default

File: REQUIRE [hunk 3]
```diff
@@ -1,5 +1,3 @@
-julia 0.5
-LibBSON 0.2.8
-BinDeps 0.4
-Compat 0.9
-@osx Homebrew
+julia 0.6
+LibBSON
+BinaryProvider
```
Comment (ScottPJones) [inline REQUIRE]: same here about the end of line.
If this doesn't work on v0.7 and above, then cap it with
julia 0.6 0.7

File: test/runtests.jl [hunk 4]
```diff
@@ -1,90 +1,108 @@
...
+@testset "Mongo" begin
     client = MongoClient()
     collection = MongoCollection(client, "foo", "bar")
     oid = BSONOID()
 
-    context("variables") do
-      @fact collection.client --> client
-      @fact collection.db --> "foo"
-      @fact collection.name --> "bar"
+    # variables
```
Comment (ScottPJones) [inline test/runtests.jl]: can use testset for the contexts of FactCheck

**Review by ScottPJones — CHANGES_REQUESTED** [review]

**Comment by felipenoris** [comment]
@ScottPJones All done. Should be good to go!

**Pull request merged**